src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -63,12 +63,136 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             DiagnosticBag diagnostics)
         {
             var type = (NamedTypeSymbol)this.BindType(node.Type, diagnostics);
+
+            var operators = type.GetMembers(WellKnownMemberNames.IsOperatorName);
```

Comment:
Shouldn't we look at base class members and extension methods as well? Perhaps we should do lookup (it will take care of accessibility checks as well)?